packages/preview-middleware/src/base/flp.ts

```diff
@@ -879,6 +926,106 @@ export class FlpSandbox {
             );
         }
     }
+
+    /**
+     * Stores the card manifest files in the webapp folder of the project and updates the application manifest.
+     *
+     * @param {Request} req - The HTTP request object containing the card manifest data in the body.
+     * @param {Response} res - The HTTP response object used to send the response back to the client.
+     * @returns {Promise<void>} A promise that resolves when the operation is complete.
+     */
+    private async storeCardManifestHandler(req: Request, res: Response): Promise<void> {
+        try {
+            const {
+                floorplan,
+                localPath,
+                fileName = FileName.Manifest,
+                manifests
+            } = req.body as {
+                floorplan: string;
+                localPath: string;
+                fileName?: string;
+                manifests: MultiCardsPayload[];
+            };
+            this.fs = this.fs ?? create(createStorage());
+            const webappPath = await getWebappPath(path.resolve(), this.fs);
+            const fullPath = join(webappPath, localPath);
+            const filePath = fileName.endsWith('.json') ? join(fullPath, fileName) : `${join(fullPath, fileName)}.json`;
+            const integrationCard = getIntegrationCard(manifests);
+            this.fs.write(filePath, JSON.stringify(integrationCard.manifest, null, 2));
+
+            const entitySet = integrationCard.entitySet;
+            const sapCardsAp = (this.manifest['sap.cards.ap'] ??= {});
+            sapCardsAp.embeds ??= {};
+            sapCardsAp.embeds[floorplan] = {
+                default: entitySet,
+                manifests: {
+                    [entitySet]: [
+                        {
+                            localUri: localPath
+                        }
+                    ]
+                }
+            } satisfies ManifestNamespace.EmbedsSettings;
+
+            const manifestPath = join(webappPath, FileName.Manifest);
+            this.fs.write(manifestPath, JSON.stringify(this.manifest, null, 2));
```

Comment:
can we change this to
```suggestion
            const appAccess = await createApplicationAccess(path.resolve(), this.fs);
            await appAccess.updateManifestJSON(this.manifest, this.fs);
```
You'll also need to 
`import { createApplicationAccess } from `@sap-ux/project-access`.
when writing JSON this way we ensure to keep existing indentation in file. 

While checking this I recognized a small bug, when passing a `memFs` instance to `createApplicationAccess()`, the call to `appAccess.updateManifestJSON()` should automatically take the passed `memFs` instance. 
I create a PR to fix this:https://github.com/SAP/open-ux-tools/pull/3221. Once the PR is in, the second statement can be simplified to
```typescript
await appAccess.updateManifestJSON(this.manifest);
```

Reply:
PR https://github.com/SAP/open-ux-tools/pull/3221 has been merged now.

Reply:
Adapted to the new changes